client/app/bundles/course/achievement/components/misc/AchievementReordering.tsx

```diff
@@ -0,0 +1,99 @@
+import { FC, useState } from 'react';
+import { defineMessages, injectIntl } from 'react-intl';
+import { Button } from '@mui/material';
+import axios from 'lib/axios';
+import { toast } from 'react-toastify';
+
+require('jquery-ui/ui/widgets/sortable');
+
+interface OwnProps {
+  intl?: any;
+}
+
+const styles = {
+  AchievementReorderingButton: {
+    background: 'white',
+    fontSize: 14,
+    marginRight: 12,
+  },
+};
+
+const translations = defineMessages({
+  startReorderAchievement: {
+    id: 'course.achievement.achievementReorder.startReorderAchievement',
+    defaultMessage: 'Start Re-ordering Achievement',
+  },
+  endReorderAchievement: {
+    id: 'course.achievement.achievementReorder.endReorderAchievement',
+    defaultMessage: 'End Re-ordering Achievement',
+  },
+  updateFailed: {
+    id: 'course.achievement.achievementReorder.updateFailed',
+    defaultMessage: 'Reorder Failed.',
+  },
+  updateSuccess: {
+    id: 'course.achievement.achievementReorder.updateSuccess',
+    defaultMessage: 'Achievements successfully reordered',
+  },
+});
+
+// Serialise the ordered achievements as data for the API call.
+function serializedOrdering(): string {
+  const options = { attribute: 'achievementid', key: 'achievement_order[]' };
+  const ordering = $('tbody').first().sortable('serialize', options);
```

Comment:
This is a rather ‘hotfix’ solution considering mui datatable cant do row drag and drop. I find it easier to do it this way since table row drag and drop is only used here. Otherwise, we may need to use another table library for this.

as of now i think it is ok to use jquery and when we have more similar tables in the future, we can then consider finding another alternative